src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/Downloader.java

```diff
@@ -43,6 +43,7 @@ void download(
       List<URL> urls,
       Map<URI, Map<String, String>> authHeaders,
       Optional<Checksum> checksum,
+      String canonicalId,
```

Comment:
The purpose and semantics of `canonical_id` is documented at https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#download

Note that this field was originally present in the PR that added the `Downloader` interface. It got _removed_ as part of the import into Google, and I just need back, because we have propagate this data to the backend somehow.